hoodie-common/src/main/java/com/uber/hoodie/common/table/HoodieTableMetaClient.java

```diff
@@ -190,13 +201,27 @@ public HoodieTableConfig getTableConfig() {
   /**
    * Get the FS implementation for this table
    */
-  public FileSystem getFs() {
+  public HoodieWrapperFileSystem getFs() {
     if (fs == null) {
-      fs = FSUtils.getFs(metaPath, hadoopConf.get());
+      FileSystem fileSystem = FSUtils.getFs(metaPath, hadoopConf.get());
+      if (fileSystem instanceof HoodieWrapperFileSystem) {
```

Comment:
when you anticipate fs to be the wrapper filesystem ?

Reply:
This was a defensive check as the effect of nesting could result in longer wait times. With  the current code, this should not happen. I can replace with a precondition that we should never see a HoodieWrapperFileSystem. Let me know how it sounds ?

Reply:
yes. We can do a PreCondition if this should not happen.. Rather hit the issue, debug and fix it, than living dangerously :)

Reply:
Added precondition.